feat(shields): Add Fifi Shield Support

hmngwy opened the pull request:
<!-- If you're adding a board/shield please fill out this check-list, otherwise you can delete it -->
## Board/Shield Check-list
 - [x] This board/shield is tested working on real hardware
 - [x] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [x] `.zmk.yml` metadata file added
 - [x] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [x] General consistent formatting of DeviceTree files
 - [x] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [x] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [x] If split, no name added for the right/peripheral half
 - [x] `.conf` file has optional extra features commented out

Nicell commented on .github/workflows/build.yml:
```diff
@@ -29,6 +29,8 @@ jobs:
           - boardsource3x4
           - corne_left
           - corne_right
+          - fifi_left
+          - fifi_right
```
Please don't edit this file for now. We're working on removing this file in favor of a dynamic build script that doesn't require explicit builds.

Nicell commented on app/boards/shields/fifi/Kconfig.defconfig:
```diff
@@ -0,0 +1,52 @@
+if SHIELD_FIFI_LEFT
```
Missing copyright header

Nicell commented on app/boards/shields/fifi/Kconfig.shield:
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2020 Pete Johanson
```
```suggestion
# Copyright (c) 2021 The ZMK Contributors
```

Nicell commented on app/boards/shields/fifi/fifi.dtsi:
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
```
```suggestion
 * Copyright (c) 2021 The ZMK Contributors
```

Nicell commented on app/boards/shields/fifi/fifi.dtsi:
```diff
@@ -0,0 +1,66 @@
+
+	};
+
+	// TODO: drop-lighting RGB node(s)?
```
```suggestion
```
This is handled by the spi  overlay

Nicell commented on app/boards/shields/fifi/fifi.keymap:
```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
```
```suggestion
 * Copyright (c) 2021 The ZMK Contributors
```

Nicell commented on app/boards/shields/fifi/fifi_left.overlay:
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
```
```suggestion
 * Copyright (c) 2021 The ZMK Contributors
```

Nicell commented on app/boards/shields/fifi/fifi_right.overlay:
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
```
```suggestion
 * Copyright (c) 2021 The ZMK Contributors
```

hmngwy commented:
@Nicell added the missing copyright lines, updated to 2022, ditched the changes to build.yml

Nicell reviewed (changes requested):
Please remove the `fifi_left.conf` and `fifi_right.conf` files! Otherwise I think this definition looks good. Thanks. You can re-request my review after updating.

petejohanson commented:
@hmngwy Please feel free to re-open this once you have a chance to address the item from @Nicell and this is ready for review again.

Thanks!

hmngwy commented:
Hi @petejohanson, rebased with main and then reapplied my changes, but I'm getting the below:

```
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake -DWEST_PYTHON=/usr/bin/python3 -B/__w/zmk/zmk/build -S/__w/zmk/zmk/app -GNinja -DBOARD=nice_nano -DSHIELD=fifi
Error: Failed to build or upload nice_nano fifi undefined
```

https://github.com/hmngwy/zmk/actions/runs/4234720633/jobs/7357427250#step:10:140

I'm not quite sure what I'm doing wrong, I checked the New Keyboard Shields guide as well if anything there is new since I slept on this PR, and it seems what I added here seems to still comply, however the error says otherwise and I can't tell what I'm missing.

If you're not too busy, a quick look would be appreciated. Thanks!

caksoylar commented:
@hmngwy You should be building with `fifi_left` and `fifi_right` shields separately, right?
